README.md

```diff
@@ -0,0 +1,73 @@
+# Overview
+
+This is the base layer for charms that wish to take advantage of
+Apache's Bigtop framework for configuring and deploying services via
+puppet. [Including this layer][building] gives you access to a Python
+class called Bigtop, which contains various useful tools, including an
+automatically run install routine.
+
+# Usage
+
+To create a charm using this base layer, first you must include it in
+your `layer.yaml` file:
+
+```yaml
+includes: ['layer:apache-bigtop-base']
+```
+
+This will fetch the layer from [interfaces.juju.solutions][] and
+incorporate it into your charm. To use the Bigtop class, import it,
+then call its .render_site_yaml and .trigger_puppet routines in
+sequence, like so:
+
+```python
+from charms.layer.apache_bigtop_base import Bigtop
+
+# Setup arguments to pass to .render_site_yaml
+hosts = {'myservice': <host>}
+roles = ['myservice-worker', 'myservice-client']
+override = {'myservice::somevalue': <somevalue>, ...}
+
```

Comment:
I wanted to give a generic example, without being too specific. Does it work to replace "worker" and "client" with something even more generic? For example:

    roles = ['myservice-servertype0', 'myservice-servertype1']